backend/src/main/java/asegroup1/api/models/landregistry/LandRegistryQuerySelect.java

```diff
@@ -1,144 +1,151 @@
+		return selectValues.get(referenceName);
 	}
 
-	/**
-	 * Select all {@link Selectable selectable's}.
-	 */
-	public void selectAll() {
-		EnumSet.allOf(Selectable.class).forEach(v -> selectableMap.add(v));
+	public String[] getSelectValues(String reference) {
+		SelectObj obj = getSelectObj(reference);
+		return obj == null ? null : new String[] { obj.referenceName, obj.aggrigation.toString(), obj.aggrigationName };
 	}
 
-	/**
-	 * Deselect the specified {@link Selectable selectable's}.
-	 * 
-	 * @param selectables to deselect
-	 */
-	public void deselect(Selectable... selectables) {
-		for (Selectable selectable : selectables) {
-			selectableMap.remove(selectable);
-		}
+	public String getSelectValueAggrigationName(String reference) {
+		SelectObj obj = getSelectObj(reference);
+		return obj == null ? null : getSelectValueAggrigationName(obj);
 	}
 
-	/**
-	 * Deselect all {@link Selectable selectable's}.
-	 */
-	public void deselectAll() {
-		selectableMap.clear();
+	public Aggrigation getSelectValueAggrigation(String reference) {
+		SelectObj obj = getSelectObj(reference);
+		return obj == null ? null : getSelectValueAggrigation(obj);
 	}
 
-	/**
-	 * Check if the specified {@link Selectable} is selected.
-	 * 
-	 * @param selectable to check
-	 * @return true, if the specified selectable is selected
-	 */
-	public boolean hasSelectable(Selectable selectable) {
-		return selectableMap.contains(selectable);
+	public static String getSelectValueAggrigationName(SelectObj reference) {
+		return reference.aggrigationName;
 	}
 
-	/**
-	 * Toggle the specified {@link Selectable}. If the specified value is selected,
-	 * it will be deselected. Otherwise it will be selected.
-	 * 
-	 * @param selectable to toggle
-	 */
-	public void toggleSelectable(Selectable selectable) {
-		if (hasSelectable(selectable)) {
-			deselect(selectable);
-		} else {
-			select(selectable);
-		}
+	public static String getSelectValueReferencenName(SelectObj reference) {
+		return reference.referenceName;
 	}
 
+	public static Aggrigation getSelectValueAggrigation(SelectObj reference) {
+		return reference.aggrigation;
+	}
 
-	public enum Selectable {
-		propertyType, estateType, transactionDate, pricePaid, newBuild, transactionCategory, paon, saon, street, locality, town, district, county, postcode;
+	public LinkedHashMap<String, SelectObj> getSelectValues() {
+		return selectValues;
 	}
 
 	/**
 	 * Builds and returns the SELECT section of the query.
 	 * 
 	 * @return the SELECT section of the query
 	 */
-	public String buildQuerySelect() {
+	public String buildQuerySelect(boolean ignoreAggrigation) {
 		StringBuilder selectStringBuilder = new StringBuilder("SELECT ");
-		for (Selectable selectable : selectableMap) {
-			selectStringBuilder.append("?" + selectable.toString() + " ");
+		for (SelectObj selectable : selectValues.values()) {
+			selectStringBuilder.append(selectable.toString(ignoreAggrigation) + " ");
```

Comment:
Should not use string concat in append argument, instead use additional append method